src/platform/silabs/wifi/SiWx/WifiInterface.cpp

```diff
@@ -187,13 +187,80 @@ constexpr uint8_t kAdvActiveScanDuration     = 15;
 constexpr uint8_t kAdvPassiveScanDuration    = 20;
 constexpr uint8_t kAdvMultiProbe             = 1;
 constexpr uint8_t kAdvScanPeriodicity        = 10;
+constexpr uint8_t kAdvEnableInstantbgScan    = 1;
 
 // TODO: Confirm that this value works for size and timing
 constexpr uint8_t kWfxQueueSize = 10;
 
 // TODO: Figure out why we actually need this, we are already handling failure and retries somewhere else.
 constexpr uint16_t kWifiScanTimeoutTicks = 10000;
 
+/**
+ * @brief Network Scan callback when the device receive a scan operation from the controller.
+ *        This callback is used whe the Network Commission Driver send a ScanNetworks command.
+ *
+ *        If the scan network was requested for a specific SSID - wfx_rsi.scan_ssid had a valid value,
+ *        the callback will only forward that specific networks information.
+ *        If no ssid is provided, wfx_rsi.scan_ssid is a nullptr, we return the information of all scanned networks.
+ */
+sl_status_t BackgroundScanCallback(sl_wifi_event_t event, sl_wifi_scan_result_t * result, uint32_t result_length, void * arg)
+{
+    SL_WIFI_ARGS_CHECK_NULL_POINTER(result);
+    VerifyOrReturnError(wfx_rsi.scan_cb != nullptr, SL_STATUS_INVALID_HANDLE);
+
+    uint32_t nbreResults = result->scan_count;
+    chip::ByteSpan requestedSsidSpan(wfx_rsi.scan_ssid, wfx_rsi.scan_ssid_length);
+
+    for (uint32_t i = 0; i < nbreResults; i++)
+    {
+        wfx_wifi_scan_result_t currentScanResult = { 0 };
+
+        // Lenght excludes null-character
+        size_t scannedSsidLenght = strnlen(reinterpret_cast<char *>(result->scan_info[i].ssid), WFX_MAX_SSID_LENGTH);
```

Comment:
I see a whole lot of `reinterpret_cast<char *>` in this pr, along with string.h functions, so I was wondering, why not use CharSpan instead of ByteSpan? Is there a hard requirement for uint8_t vs char somewhere?

Reply:
Wi-Fi SDK doesn't manage the SSID the same way as an input and as an ouput so we can't have a uniform way across the board for the time being. 

Inputs are uint8_t[] with the size (without the '\0') and the outputs are also uin8t but without the size but with the '\0'.

Also, matter uses `ByteSpan` to manages ssids. As such, i think we should use the same thing.